lib/mix/lib/mix/cli.ex

```diff
@@ -115,6 +116,17 @@ defmodule Mix.CLI do
     end
   end
 
+  defp change_target(task) do
+    if target = preferred_cli_target(task) do
+      Mix.target(target)
+
+      if project = Mix.Project.pop() do
+        %{name: name, file: file} = project
+        Mix.Project.push(name, file)
```

Comment:
Because this operation busts a bunch of cache, I think we should unify change_env and change_target into a single function and pop+push the project only once.

Reply:
I updated this to only reload the project once if either of the two execute.